src/executables/compactdb_common.c

```diff
@@ -0,0 +1,286 @@
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ */
+
+
+#include "config.h"
+
+#include <stdio.h>
+#include <errno.h>
+#include <string.h>
+#include "porting.h"
+#include "dbtype.h"
+#include "dbdef.h"
+#include "load_object.h"
+#include "db.h"
+#include "locator_cl.h"
+#include "locator_sr.h"
+#include "schema_manager.h"
+#include "heap_file.h"
+#include "system_catalog.h"
+#include "object_accessor.h"
+#include "set_object.h"
+#include "btree.h"
+#include "message_catalog.h"
+#include "network_interface_cl.h"
+#include "server_interface.h"
+#include "system_parameter.h"
+#include "utility.h"
+#include "authenticate.h"
+#include "transaction_cl.h"
+#include "execute_schema.h"
+
+int get_class_mops (char **class_names, int num_class,
+						   MOP ** class_list, int *num_class_list);
+int get_class_mops_from_file (const char *input_filename, MOP ** class_list,
+									 int *num_class_mops);
+
+/*
+ * get_num_requested_class - Get the number of class from specified
+ * input file
+ *    return: error code
+ *    input_filename(in): input file name
+ *    num_class(out) : pointer to returned number of classes
+ */
+int
+get_num_requested_class (const char *input_filename, int *num_class)
+{
+  FILE *input_file;
+  char buffer[DB_MAX_IDENTIFIER_LENGTH];
+
+  if (input_filename == NULL || num_class == NULL)
+    {
+      return ER_FAILED;
+    }
+
+  input_file = fopen (input_filename, "r");
+  if (input_file == NULL)
+    {
+      perror (input_filename);
+      return ER_FAILED;
+    }
+
+  *num_class = 0;
+  while (fgets ((char *) buffer, DB_MAX_IDENTIFIER_LENGTH,
+        input_file) != NULL)
+    {
+      (*num_class)++;
+    }
+
+  fclose (input_file);
+
+  return NO_ERROR;
+}
+
+/*
+ * get_class_mops - Get the list of mops of specified classes
+ *    return: error code
+ *    class_names(in): the names of the classes
+ *    num_class(in): the number of classes
+ *    class_list(out): pointer to returned list of mops
+ *    num_class_list(out): pointer to returned number of mops
+ */
+int
+get_class_mops (char **class_names, int num_class,
+		MOP ** class_list, int *num_class_list)
+{
+  int i, status = NO_ERROR;
+  char downcase_class_name[SM_MAX_IDENTIFIER_LENGTH];
+  DB_OBJECT *class_ = NULL;
+
+  if (class_names == NULL || num_class <= 0 || class_list == NULL ||
+      num_class_list == NULL)
+    {
+      return ER_FAILED;
+    }
+
+  *num_class_list = 0;
+  *class_list = (DB_OBJECT **) malloc (DB_SIZEOF (DB_OBJECT *) * (num_class));
+  if (*class_list == NULL)
+    {
+      return ER_FAILED;
+    }
+
+  for (i = 0; i < num_class; ++i)
+    {
+      (*class_list)[i] = NULL;
+    }
```

Comment:
Is initialization necessary?

Reply:
This code is need for protecting abnormal free,  when occur error and release memory.